### PR TITLE
 Test: Fixes and improvements in end2end_mocked tests and zhmcclient mock support

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: tests/end2end/mocked_inventory.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#         vars: ...
+#         children: ...
+
+all:
+  hosts:
+    mocked_hmc_z16:
+      description: "Mocked HMC with z16 in classic mode and z16 in DPM mode"
+      mock_file: "mocked_hmc_z16.yaml"
+      cpcs:
+        CPC1:
```
Comment (EdGue42) [inline tests/end2end/mocked_inventory.yaml:40]: minor: why not name them "CLASSIC" and "DPM" instead just 1 and 2 ?

File: tests/end2end/test_storage_volume.py [hunk 2]
```diff
@@ -203,6 +203,9 @@ def test_stovol_crud(dpm_mode_cpcs):  # noqa: F811
             stovol = stogrp.storage_volumes.create(stovol_input_props)
 
             for pn, exp_value in stovol_input_props.items():
+                assert pn in stovol.properties, (
+                    f"Input property {pn!r} is not in created storage volume:\n"
+                    f"{stovol!r}")
```
Comment (EdGue42) [inline tests/end2end/test_storage_volume.py:208]: consistency: you are using the new f"{}" style here, but keep the old style in other places? 
Maybe run that "update" script on the files you modify?

**Review by EdGue42 — APPROVED** [review]

File: tests/end2end/mocked_inventory.yaml (hunk 1 above)
Comment (andy-maier) [reply tests/end2end/mocked_inventory.yaml]: Good idea. Done (Used CPC_DPM and CPC_CLA)

File: tests/end2end/test_storage_volume.py (hunk 2 above)
Comment (andy-maier) [reply tests/end2end/test_storage_volume.py:208]: For reasons I do not understand yet, the pyupgrade tool I used for the upgrade of the code to Python 3.8 converted only a subset of the format strings to f-strings. I would like to convert the remaining ones in a separate PR.

**Comment by andy-maier** [comment]
Updated to reflect the review comments.

**Review by EdGue42 — APPROVED** [review]

**Pull request merged**